### ISSUE TITLE
Memory usage in long run

### ISSUE THREAD

**Issue by aik099** [padraic/mockery#239]
From my observations the Mockery seems to store all created mock instances (or at least their expectations) somewhere internally. This can cause problems in large tests, when after mock is no longer needed in one of the tests it's not being garbage collected, because Mockery container has reference to it.

I think, that we can safely assume, that by the time, when Mockery test listener is called by PHPUnit (in `tearDownAfterClass`), then we can free memory by removing unused mocks.

If somebody wants to test, then feel free to grab https://github.com/aik099/qa-tools, which test suite is using Mockery extensivery and can eat up till 300MB of memory.

This was too much for TravisCI and I decided to run tests in parallel to make each process memory consumption smaller.
